caddy.go

```diff
@@ -241,6 +241,9 @@ func listenerAddrEqual(ln net.Listener, addr string) bool {
 	if lnAddr == net.JoinHostPort("0.0.0.0", port) {
 		return true
 	}
+	if hostname != "" && lnAddr == addr {
+		return true
+	}
 	return false
```

Comment:
These last 4 lines can be replaced with simply: `return hostname != "" && lnAddr == addr`